testtools/matchers/_warnings.py

```diff
@@ -0,0 +1,97 @@
+# Copyright (c) 2009-2016 testtools developers. See LICENSE for details.
+
+__all__ = [
+    'Warnings',
+    'warning_message',
+    'is_deprecated']
+
+import warnings
+
+from ._basic import Is
+from ._const import Always
+from ._datastructures import MatchesListwise, MatchesStructure
+from ._higherorder import (
+    AfterPreprocessing,
+    Annotate,
+    MatchesAll,
+    Not,
+    )
+from ._impl import Mismatch
+
+
+def warning_message(category_type, message=None, filename=None, lineno=None,
```

Comment:
@rbtcollins @jml : did we ever settle on a standard naming convention for matcher types? Having some matchers as functions and some as classes is a little irksome. I'm not sure what we should do at this point:

 * Follow pep8, and force users to import a mix of `function_name` and `ClassName` matchers?
 * Make functions follow `ClassName` convention (FWIW, we do this in `testtools.content` a fair bit).

Thoughts?